Vesper-Agents-Suite/xy-safe-borrow-agent/src/agent.ts

```diff
@@ -0,0 +1,166 @@
+import { STRATEGY_ABI, CTOKEN_ABI, ORACLE_ABI, TRADER_JOE_ABI, COMPOUND_ABI, POOL_ABI } from "./abi";
+import { createFinding } from "./utils";
+import BigNumber from "bignumber.js";
+const XY_STRATEGY_KEY = "_compoundXYStrategies"
+const XY_STRATEGY_NAME = "XYStrategy"
+const Aave = "Aave"
+const TRADER_JOE = 'TraderJoe'
+const web3: Web3 = new Web3(getJsonRpcUrl());
+const cache: LRU<string, string[]> = new LRU<string, string[]>({ max: 100 });
+const tracker: TimeTracker = new TimeTracker();
+const vesperFetcher: VesperFetcher = new VesperFetcher(web3);
+let cacheTime: number = 3600000; // one hour in milliseconds
+const DECIMAL18 = new BigNumber('1e18')
+const MAX_BPS = 10000
+
+export const setCacheTime = (time: number) => {
+  cacheTime = time;
+}
+
+const fetchXYStrategies = async (
+  blockNumber: string
+): Promise<string[]> => {
+  if (blockNumber != "latest" && cache.get(blockNumber) !== undefined)
+    return cache.get(blockNumber) as string[];
+  let [V3] = await Promise.all([
+    vesperFetcher.getStrategiesV3(blockNumber)
+  ]);
+  V3 = Array.from(new Set<string>(V3));
+  const valueV3promise = V3.map(async (strategy: string) => {
+    const strategyContract = new web3.eth.Contract(
+      STRATEGY_ABI,
+      strategy,
+    );
+
+    const name: string = await strategyContract.methods.NAME().call({}, blockNumber);
+    // Compound or it's forked XY Strategies 
+    if (!name.includes(XY_STRATEGY_NAME) || name.includes(Aave)) return BigInt(0);
+
+    // get borrow token
+    const borrowCTokenAddress: string = await strategyContract.methods.borrowCToken().call({}, blockNumber);
+    const borrowCToken = new web3.eth.Contract(
+      CTOKEN_ABI,
+      borrowCTokenAddress,
+    );
+
+    // get supply token
+    const supplyCTokenAddress: string = await strategyContract.methods.supplyCToken().call({}, blockNumber);
+    const supplyCToken = new web3.eth.Contract(
+      CTOKEN_ABI,
+      supplyCTokenAddress,
+    );
+
+    // get comptroller
+    const comptrollerAddress: string = await strategyContract.methods.comptroller().call({}, blockNumber);
+    const actualComptrollerAbi = name.includes(TRADER_JOE) ? TRADER_JOE_ABI : COMPOUND_ABI
+    const comptrollerContract = new web3.eth.Contract(
+      actualComptrollerAbi,
+      comptrollerAddress,
+    );
+
+    // get Oracle
+    const oracleAddress: string = await comptrollerContract.methods.oracle().call({}, blockNumber);
+    const oracleContract = new web3.eth.Contract(
+      ORACLE_ABI,
+      oracleAddress,
+    );
+
+    // get prices
+    const borrowTokenPrice = new BigNumber(await oracleContract.methods.getUnderlyingPrice(borrowCTokenAddress).call({}, blockNumber));
+    const supplyTokenPrice = new BigNumber(await oracleContract.methods.getUnderlyingPrice(supplyCTokenAddress).call({}, blockNumber));
+    const borrowed = new BigNumber(await borrowCToken.methods.borrowBalanceStored(strategy).call({}, blockNumber));
```

Comment:
why do you need to create BigNumber?  isnt borrowCToken.methods.borrowBalanceStored(strategy).call return bigNumber()?